### ISSUE TITLE
export.py - TensorFlow Lite export failure due to EndVector()

### ISSUE THREAD

**Issue by HripsimeS** [ultralytics/yolov5#10202]
### Search before asking

- [X] I have searched the YOLOv5 [issues](https://github.com/ultralytics/yolov5/issues) and [discussions](https://github.com/ultralytics/yolov5/discussions) and found no similar questions.

### Question

@glenn-jocher Hello. I need to convert yolov5s.pt weight to yolov5s.tflite and save tflite in some folder. I saw that export.py is normally doing that action, but I did not receive the outcome model as tflite format. I run this command:

**python export.py --weights yolov5s.pt  --include tflite**

In **weights** folder it saved **best_saved_model** folder with 
1) **assets** folder which was empty, 
2) **variables** folder with  **variables.data-00000-of-00001** and **variables.index** files; 
3) **saved_model.pb** file.

Tha last line after running the command above are these 
**TensorFlow Lite: export failure  58.0s: EndVector() takes 1 positional argument but 2 were given**

So the export was failed due to EndVector(), which I could not find in export.py file. Could you please help me to figure out what is the issue and how to fix it in order to get an outcome **yolov5s.tflite** file. Look forward to hearing soon!

**Comment by glenn-jocher** [comment]
@HripsimeS 👋 Hello! Thanks for asking about YOLOv5 🚀 **benchmarks**. YOLOv5 inference is officially supported in 11 formats, and all formats are benchmarked for identical accuracy and to compare speed every 24 hours by the [YOLOv5 CI](https://github.com/ultralytics/yolov5/actions/workflows/ci-testing.yml?query=event%3Aschedule).

Due to these daily benchmarks we can tell that TFLite export is operating correctly and there are no errors there.

💡 ProTip: Export to ONNX or OpenVINO for up to 3x CPU speedup. See [CPU Benchmarks](https://github.com/ultralytics/yolov5/pull/6613).
💡 ProTip: Export to TensorRT for up to 5x GPU speedup. See [GPU Benchmarks](https://github.com/ultralytics/yolov5/pull/6963).

Format                      | `export.py --include`         | Model
:---                        | --:                           | :--
[PyTorch](https://pytorch.org/)                     | -                             | `yolov5s.pt`
[TorchScript](https://pytorch.org/docs/stable/jit.html)                 | `torchscript`                 | `yolov5s.torchscript`
[ONNX](https://onnx.ai/)                        | `onnx`                        | `yolov5s.onnx`
[OpenVINO](https://docs.openvino.ai/latest/index.html)                    | `openvino`                    | `yolov5s_openvino_model/`
[TensorRT](https://developer.nvidia.com/tensorrt)                    | `engine`                      | `yolov5s.engine`
[CoreML](https://github.com/apple/coremltools)                      | `coreml`                      | `yolov5s.mlmodel`
[TensorFlow SavedModel](https://www.tensorflow.org/guide/saved_model)       | `saved_model`                 | `yolov5s_saved_model/`
[TensorFlow GraphDef](https://www.tensorflow.org/api_docs/python/tf/Graph)         | `pb`                          | `yolov5s.pb`
[TensorFlow Lite](https://www.tensorflow.org/lite)             | `tflite`                      | `yolov5s.tflite`
[TensorFlow Edge TPU](https://coral.ai/docs/edgetpu/models-intro/)         | `edgetpu`                     | `yolov5s_edgetpu.tflite`
[TensorFlow.js](https://www.tensorflow.org/js)               | `tfjs`                        | `yolov5s_web_model/`


## Benchmarks

Benchmarks below run on a Colab Pro with the YOLOv5 tutorial notebook <a href="https://colab.research.google.com/github/ultralytics/yolov5/blob/master/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a>. To reproduce:
```bash
python utils/benchmarks.py --weights yolov5s.pt --imgsz 640 --device 0
```

### Colab Pro V100 GPU

```
benchmarks: weights=/content/yolov5/yolov5s.pt, imgsz=640, batch_size=1, data=/content/yolov5/data/coco128.yaml, device=0, half=False, test=False
Checking setup...
YOLOv5 🚀 v6.1-135-g7926afc torch 1.10.0+cu111 CUDA:0 (Tesla V100-SXM2-16GB, 16160MiB)
Setup complete ✅ (8 CPUs, 51.0 GB RAM, 46.7/166.8 GB disk)

Benchmarks complete (458.07s)
                   Format  mAP@0.5:0.95  Inference time (ms)
0                 PyTorch        0.4623                10.19
1             TorchScript        0.4623                 6.85
2                    ONNX        0.4623                14.63
3                OpenVINO           NaN                  NaN
4                TensorRT        0.4617                 1.89
5                  CoreML           NaN                  NaN
6   TensorFlow SavedModel        0.4623                21.28
7     TensorFlow GraphDef        0.4623                21.22
8         TensorFlow Lite           NaN                  NaN
9     TensorFlow Edge TPU           NaN                  NaN
10          TensorFlow.js           NaN                  NaN
```

### Colab Pro CPU

```
benchmarks: weights=/content/yolov5/yolov5s.pt, imgsz=640, batch_size=1, data=/content/yolov5/data/coco128.yaml, device=cpu, half=False, test=False
Checking setup...
YOLOv5 🚀 v6.1-135-g7926afc torch 1.10.0+cu111 CPU
Setup complete ✅ (8 CPUs, 51.0 GB RAM, 41.5/166.8 GB disk)

Benchmarks complete (241.20s)
                   Format  mAP@0.5:0.95  Inference time (ms)
0                 PyTorch        0.4623               127.61
1             TorchScript        0.4623               131.23
2                    ONNX        0.4623                69.34
3                OpenVINO        0.4623                66.52
4                TensorRT           NaN                  NaN
5                  CoreML           NaN                  NaN
6   TensorFlow SavedModel        0.4623               123.79
7     TensorFlow GraphDef        0.4623               121.57
8         TensorFlow Lite        0.4623               316.61
9     TensorFlow Edge TPU           NaN                  NaN
10          TensorFlow.js           NaN                  NaN
```

Good luck 🍀 and let us know if you have any other questions!

**Comment by HripsimeS** [comment]
@glenn-jocher thanks for your quick reply, I could fix it with this command line, I will close this issue!
**pip install --upgrade flatbuffers==1.12** # downgrade from v2 to v1.12

**Comment by glenn-jocher** [comment]
Oh that's strange. We used to install a fixed version of flatbuffers like this but it caused problems with newer versions of TF I think.

**Comment by HripsimeS** [comment]
@glenn-jocher Hello. I trained my dataset for yolov5s model, already tested it with detect.py and it works fine as **pt** format.
Recently I converted my trained yolov5s **pt** model to **tflite** format by using a command below:
**python export.py --weights yolov5s.pt --include tflite**

Then I used a basic template to deploy my object detection model on Android Studio by following the instruction below: https://developers.google.com/codelabs/tflite-object-detection-android

The app works perfect with all 7 **tflite** models mentioned below
https://tfhub.dev/tensorflow/collections/lite/task-library/object-detector/1

But I got an error while using my own trained  yolov5s model which I converted from **pt** to **tflite** format using **export.py**. Probably in this case I can conclude that the issue is in converting process while using **export.py** and that's why my model in **tflite** format does not work at all in Android studio and app goes down when I try to do a detection.
In attached you can find a report bug description while I was trying the app with my converted model.

Do you have any ideas how to verify if the issue is connected with my converted/exported **tflite** model or it is connected with something else?  Thanks in advance!


![image](https://user-images.githubusercontent.com/42246765/203036047-8f725634-8bd0-48fd-9eee-bd197f11fc75.png)
